File: example/GdeckObjC/ViewController.m
//
//  ViewController.m
//  GdeckObjC
//
//  Created by Rohit Sangwan on 13/10/23.
//

#import "ViewController.h"
@import GesturedeckiOS;

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIButton *btnStart;
@property (weak, nonatomic) IBOutlet UIButton *btnStop;

@property (strong, nonatomic) GesturedeckMedia *gesturedeckMedia;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad]; 
    self.gesturedeckMedia = [[GesturedeckMedia alloc] init];
}

//- (IBAction)onStartButtonTap:(UIButton *)sender {
//    [self.gesturedeckMedia start];
//}
//
//- (IBAction)onStopButtonTap:(UIButton *)sender {
//    [self.gesturedeckMedia stop];
//}
@end

File: example/GdeckUIKit/ViewController.swift
//
//  ViewController.swift
//  GdeckUIKit
//
//  Created by Rohit Sangwan on 13/10/23.
//

import GesturedeckiOS
import UIKit

class ViewController: UIViewController {
    @IBOutlet var btnStart: UIButton!
    @IBOutlet var btnStop: UIButton!
    @IBOutlet var txtGestureEvent: UILabel!

    var sceneDelegate: SceneDelegate? = nil

    override func viewDidLoad() {
        super.viewDidLoad()
        sceneDelegate = UIApplication.shared.connectedScenes.first?.delegate as? SceneDelegate
        btnStart.addTarget(self, action: #selector(onStartButtonTap(_:)), for: .touchUpInside)
        btnStop.addTarget(self, action: #selector(onStopButtonTap(_:)), for: .touchUpInside)
    }

    @IBAction func onStartButtonTap(_: UIButton) {
        sceneDelegate?.gesturedeckMedia?.start()
    }

    @IBAction func onStopButtonTap(_: UIButton) {
        sceneDelegate?.gesturedeckMedia?.stop()
    }
}
